scripts/getXcnRate.js

```diff
@@ -0,0 +1,82 @@
+const axios = require('axios');
+
+const BASE_URL = 'https://api.onyx.org/api';
+const XCNContractAddress = '0x1961AD247B47F4f2242E55a0E5578C6cf01F8D12';
```

Comment:
Everything looks good to me.
The only thing is that the name of `XCNContractAddress`.
Actually that address is oXCN token, and the name - `XCNContractAddress` looks like the XCN token address.

Reply:
Yes, it's oXCN address. API responds with empty result when XCN address is used. `getXcnRate()`: `tokenPrice` is price of XCN token, not oXCN. `getXcnRateByBlockNumber()`: `priceUSD` is price of XCN token, not oXCN.